File: SDE_sheet/Arrays/21_longest_consecutive_sequence/solution.js
var longestConsecutive = function(nums) {
    let max = 0;
    const set = new Set(nums);
    for(let num of set){
        if(!set.has(num - 1)){
            let curr = 1;
            while(set.has(num + 1)){
                num++;
                curr++;
            }
            max = Math.max(max, curr);
        }
    }
    return max;
};

File: SDE_sheet/Arrays/21_longest_consecutive_sequence/solution.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }

        int max = 0;
        for (int num : set) {
            if (!set.contains(num - 1)) {
                int curr = 1;
                while (set.contains(num + 1)) {
                    num++;
                    curr++;
                }
                max = Math.max(max, curr);
            }
        }
        return max;
    }
}
